libafl_frida/src/lib.rs

```diff
@@ -344,3 +344,153 @@ impl Default for FridaOptions {
+            > = lib.get(test.as_bytes()).unwrap();
+
+            let harness = |input: &BytesInput| {
+                let target = input.target_bytes();
+                let buf = target.as_slice();
+                (target_func)(buf.as_ptr(), buf.len());
+                ExitKind::Ok
+            };
+
+            // This actually should check for 1, but as of now we get 70
+            assert!(test_asan_with_harness(harness, options) > 0);
+        }
+    }
+
+    unsafe fn test_asan_with_harness<F>(mut harness: F, options: &FuzzerOptions) -> usize
+    where
+        F: FnMut(&BytesInput) -> ExitKind,
+    {
+        // let gum = Gum::obtain();
+
+        let mut corpus = InMemoryCorpus::<BytesInput>::new();
+
+        //TODO - make sure we use the right one
+        let testcase = Testcase::new(vec![0; 4].into());
+        corpus.add(testcase).unwrap();
+
+        let coverage = CoverageRuntime::new();
+        let asan = AsanRuntime::new(&options);
+        let mut frida_helper = FridaInstrumentationHelper::new(
+            /*&gum*/ &GUM,
+            &options,
+            tuple_list!(coverage, asan),
+        );
+
+        let rand = StdRand::with_seed(0);
+
+        let mut feedback = ConstFeedback::new(false);
+
+        // Feedbacks to recognize an input as solution
+        let mut objective = feedback_or_fast!(
+            // true enables the AsanErrorFeedback
+            feedback_and_fast!(ConstFeedback::from(true), AsanErrorsFeedback::new())
+        );
+
+        let mut state = StdState::new(
+            rand,
+            corpus,
+            InMemoryCorpus::<BytesInput>::new(),
+            &mut feedback,
+            &mut objective,
+        )
+        .unwrap();
+
+        let mut event_manager = NopEventManager::new();
+
+        let mut fuzzer = StdFuzzer::new(StdScheduler::new(), feedback, objective);
+
+        let observers = tuple_list!(
+            AsanErrorsObserver::new(&ASAN_ERRORS) //,
+        );
+
+        let mut executor = FridaInProcessExecutor::new(
+            &GUM, /*&gum,*/
+            InProcessExecutor::new(
+                &mut harness,
+                observers, // tuple_list!(),
+                &mut fuzzer,
+                &mut state,
+                &mut event_manager,
+            )
+            .unwrap(),
+            &mut frida_helper,
+        );
+
+        let mutator = StdScheduledMutator::new(tuple_list!(BitFlipMutator::new()));
+        let mut stages = tuple_list!(StdMutationalStage::new(mutator));
+
+        // log::info!("Starting fuzzing!");
+        fuzzer
+            .fuzz_one(&mut stages, &mut executor, &mut state, &mut event_manager)
+            .unwrap_or_else(|_| panic!("Error in fuzz_one"));
+
+        // log::info!("Done fuzzing! Got {} solutions", state.solutions().count());
+        state.solutions().count()
+    }
+
+    #[test]
+    #[cfg(unix)]
+    fn run_test_asan() {
+        env_logger::init();
```

Comment:
The ability to specify the trace level at runtime is very helpful. Does SimpleStdoutLogger support this?

Reply:
Moved to dev-dependencies

Reply:
Not sure, but for a test you don't usually see the logs anyway? For development purposes it's a different story, but I'd still prefer to not add crates to the library that are not needed in the normal case

Reply:
FWIW I usually use something like this to enable trace in debug builds
```rust
    if cfg!(debug_assertions) {
        log::set_max_level(log::LevelFilter::Trace);
    } else {
        log::set_max_level(log::LevelFilter::Info);
    }
    SimpleStdoutLogger.set_logger().unwrap()
```

Reply:
I indeed use this while developing the test, because it is quite complicated. I can remove this altogether or switch to SimpleStdoutLogger if it is important. However, I don't see it used anywhere as of now

Reply:
I've written the code using SimpleStdoutLogger, but the prints don't appear even though writing to stdout stream works...

Reply:
That's somewhat concerning, you did set the LeveFilter accordingly? Do rust tests close stdout or something?

Reply:
I found the problem. In cargo test, the output of successfull tests is suppressed by default, both those sent to stdout and stderr. To see the output, run `cargo test -- --nocapture`. Pushed the code